pkg/storage/gcs/gcs.go

```diff
@@ -283,7 +283,8 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 
 			if err != nil {
 				if gerr, ok := err.(*gapi.Error); ok {
-					util.UpdateCondition(cr, imageregistryv1.StorageEncrypted, operatorapi.ConditionFalse, strconv.Itoa(gerr.Code), gerr.Error())
+					util.UpdateCondition(cr, imageregistryv1.StorageEncrypted, operatorapi.ConditionFalse, "InvalidStorageConfiguration", gerr.Error())
+					return err
```

Comment:
We can probably be a bit more intelligent here about the reason - ex: a 5xx error does not imply that we sent bad configuration to GCP.